radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/P2PModule.java

```diff
@@ -127,9 +128,21 @@ protected void configure() {
   }
 
   @ProvidesIntoSet
-  private EventProcessorOnRunner<?> peerEventProcessor(PeerManager peerManager) {
+  private EventProcessorOnRunner<?> peerEventProcessor(
+      PeerManager peerManager, ProxiedPeersViewService proxiedPeersViewService) {
     return new EventProcessorOnRunner<>(
-        Runners.P2P_NETWORK, PeerEvent.class, peerManager.peerEventProcessor());
+        Runners.P2P_NETWORK,
+        PeerEvent.class,
+        fanOut(
```

Comment:
:+1: This is an interesting approach, I think we could apply it to a few other modules as well (not in this PR).
So far we've been exposing separate `EventProcessorOnRunner` for each EventProcessor (even within a single Runner), for example:
```
private EventProcessorOnRunner<?> peerEventProcessor_peerManager(PeerManager peerManager) { ... }
private EventProcessorOnRunner<?> peerEventProcessor_proxiedPeersViewService(ProxiedPeersViewService proxiedPeersViewService) { ... }
```